src/instantiator.js

```diff
@@ -105,6 +105,13 @@ function instantiate(schema, options) {
       return;
     }
     var i;
+    // Check if type is array of types
+    // if so, set type to first in list of types
+    if (Array.isArray(obj.type)) {
```

Comment:
Ok I moved the function up and out, although instead of returning the type which I think is what you were trying to get at with `var type = getObjectType(obj);` I'm checking for an array of types.

I'm trying to check if you've passed an array of types, though if obj.type isn't overridden from the array that was passed in it will the the isPrimitive check.

So I overwrite the obj.type to the first type in the array of types. There's probably a better way to handle this, but I was looking to make the minimal changes to the lib, until I got some feedback and had a better understanding of the JSONschema spec.